app/assets/javascripts/index/directions.js

```diff
@@ -82,17 +78,31 @@ OSM.Directions = function (map) {
     input.on("change", function (e) {
       // make text the same in both text boxes
       var value = e.target.value;
-      endpoint.setValue(value);
+      endpoint.setValue(value, false, null, true);
     });
 
-    endpoint.setValue = function (value, latlng) {
+    endpoint.setValue = function (value, force_reverse_geocoding, latlng, fitMap) {
```

Comment:
Why would a single endpoint control fitting?

Then below you have:

    endpoints[0].setValue(params.from || "", !ll_from_search_form, from, false);
    endpoints[1].setValue(params.to || "", !ll_from_search_form, to, true);

And not it's not clear why the first endpoint gets `fitMap = false` but the second one gets `fitMap = true`.

Reply:
> Why would a single endpoint control fitting?
> 

Because the route can change on single endpoint update, so, it can go out of viewing area and we need to fit bounds.

> Then below you have:
> 
> ```
> endpoints[0].setValue(params.from || "", !ll_from_search_form, from, false);
> endpoints[1].setValue(params.to || "", !ll_from_search_form, to, true);
> ```
> 
> And not it's not clear why the first endpoint gets `fitMap = false` but the second one gets `fitMap = true`.

Because in some situations there is a flickering if we are doing fitting both times.

Reply:
>> Why would a single endpoint control fitting?

> Because the route can change on single endpoint update

It's for the code outside of endpoints to determine what to do with the route.